pkg/controller/jobs/pod/pod_controller.go

```diff
@@ -667,9 +667,38 @@ func (p *Pod) Load(ctx context.Context, c client.Client, key *types.NamespacedNa
 }
 
 func (p *Pod) constructGroupPodSets() ([]kueue.PodSet, error) {
+	if _, useFastAdmission := p.pod.GetAnnotations()[GroupFastAdmissionAnnotation]; useFastAdmission {
+		tc, err := p.groupTotalCount()
+		if err != nil {
+			return nil, err
+		}
+		return constructGroupPodSetsFast(p, tc)
+	}
 	return constructGroupPodSets(p.list.Items)
 }
 
+func constructGroupPodSetsFast(p *Pod, groupTotalCount int) ([]kueue.PodSet, error) {
+	podSets := make([]kueue.PodSet, 1)
+	for i, podInGroup := range p.list.Items {
+		if !isPodRunnableOrSucceeded(&podInGroup) {
+			continue
+		}
+
+		roleHash, err := getRoleHash(podInGroup)
+		if err != nil {
+			return nil, fmt.Errorf("failed to calculate pod role hash: %w", err)
+		}
+
+		podSet := FromObject(&podInGroup).PodSets()
+		podSet[0].Name = roleHash
+		podSet[0].Count = int32(groupTotalCount)
+		podSets[i] = podSet[0]
+		return podSets, nil
+	}
```

Comment:
We should make sure we are creating at most one podset with the count: groupTotalCount.

```suggestion
	for i, podInGroup := range p.list.Items {
		if !isPodRunnableOrSucceeded(&podInGroup) {
			continue
		}

		roleHash, err := getRoleHash(podInGroup)
		if err != nil {
			return nil, fmt.Errorf("failed to calculate pod role hash: %w", err)
		}

		podSets := FromObject(&podInGroup).PodSets()
		podSets[0].Name = roleHash
		podSets[0].Count = int32(groupTotalCount)
		return podSets, nil
	}
```